src/AddressRegistry.Api.Legacy/AddressMatch/Matching/Sanitizer.cs

```diff
@@ -5,376 +5,581 @@ namespace AddressRegistry.Api.Legacy.AddressMatch.Matching
-        private static HashSet<string> _indexWordsFloor = new HashSet<string>(StringComparer.InvariantCultureIgnoreCase) { "RC", "RCAR", "GV", "Rch", "RDC", "HAL", "rez", "RCH", "GV", "GVL", "GLV", "GEL", "GL" };
-        private static HashSet<string> _indexWordsWithoutBisNumber = new HashSet<string>(_indexWordsBoxNumber.Concat(_indexWordsAppartementNumber).Concat(_indexWordsFloorNumber).Concat(_indexWordsFloor), StringComparer.InvariantCultureIgnoreCase);
         public IWarningLogger Warnings { get; set; }
 
         public Sanitizer()
-        {
-            Warnings = new DefaultWarningLogger();
-        }
+            => Warnings = new DefaultWarningLogger();
 
-        public List<HouseNumberWithSubaddress> Sanitize(string streetName, string houseNumber, string index)
+        public List<HouseNumberWithSubaddress> Sanitize(
+            string? streetName,
+            string? houseNumber,
+            string? index)
         {
-            //TODO: clean up Sammy code
-            int hnr, bnr;
-
-            //huisnummer formatteren
-            var houseNumber0 = FormatRrHouseNumber(houseNumber);
-            if (houseNumber0 == "00000000")
+            // huisnummer formatteren
+            var formattedHouseNumber = FormatRrHouseNumber(houseNumber);
+            if (formattedHouseNumber == "00000000")
                 Warnings.AddWarning("1", "Ongeldig 'Huisnummer'.");
 
-            //rrindex formatteren
-            var rrindex0 = FormatRrIndex(index);
+            // rrindex formatteren
+            var formattedRrIndex = FormatRrIndex(index);
 
-            /* huisnummer in straatnaam */
-            if ((string.IsNullOrEmpty(houseNumber) || houseNumber0 == "00000000") && string.IsNullOrEmpty(index) && !string.IsNullOrEmpty(streetName))
+            // huisnummer in straatnaam
+            if ((string.IsNullOrEmpty(houseNumber) || formattedHouseNumber == "00000000") &&
+                string.IsNullOrEmpty(index) &&
+                !string.IsNullOrEmpty(streetName))
             {
-                var requestStraatnaamWithoutStraatnaam = streetName.Replace(StripStreetName(streetName), "").Replace(",", string.Empty).Replace(" ", string.Empty);
-                //huisnummer als prefix in straatnaam
+                int hnr;
+
+                var requestStraatnaamWithoutStraatnaam = streetName
+                    .Replace(StripStreetName(streetName), string.Empty)
+                    .Replace(",", string.Empty)
+                    .Replace(" ", string.Empty);
+
+                var possibleNumberInStreet = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
+                var possibleIndexInStreet = FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam));
+
+                // huisnummer als prefix in straatnaam
+                // TODO: [0-9]+ ?
                 if (new Regex("[0-9] ").IsMatch(streetName))
```

Comment:
Shouldn't this be `[0-9]+` instead?

Reply:
In retrospect, this will match "X " so "blablaXX " is seen as a match too :)

It would still be cleaner with `[0-9]+` explicitly.